packages/modules/vehicles/mercedeseq/api.py

```diff
@@ -0,0 +1,209 @@
+        socDebugLog("Conf Access Tok: " + access_token)
+        socDebugLog("Conf Refresh Tok: " + refresh_token)
+        socDebugLog("Conf Expires_in: " + str(expires_in))
+
+    # socDebugLog("Token expires at: " +time.strftime( "%d.%m.%Y  %H:%M:%S",time.localtime(expires_in)))
+    socDebugLog("Token expires in: " + str(int(expires_in) - int(time.time())) + "s. at: " +
+                time.strftime("%d.%m.%Y  %H:%M:%S", time.localtime(expires_in)))
+
+    if int(expires_in) < int(time.time()):
+        # Access Token is exired
+        if Debug >= 1:
+            socDebugLog("Acc Token Expired")
+
+        # get new Access Token with referesh token
+        data = {'grant_type': 'refresh_token', 'refresh_token': refresh_token}
+        ref = requests.post(tok_url, data=data, verify=True, allow_redirects=False, auth=(client_id, client_secret),
+                            timeout=req_timeout)
+        if Debug >= 1:
+            socDebugLog("Refresh Token Call:" + str(ref.status_code))
+            socDebugLog("Refresh Token Text:" + str(ref.text))
+
+        # write HTTP reponse code to file
+        # try:
+        #     fd = open(ramdiskdir + 'soc_eq_lastresp','w')
+        #     fd.write(str(ref.status_code))
+        #     fd.close()
+        # except:
+        #     fd.close()
+
+        if ref.status_code == 200:
+            # valid response
+            tok = json.loads(ref.text)
+
+            access_token = tok['access_token']
+            refresh_token = tok['refresh_token']
+            expires_in = tok['expires_in'] - 60 + int(time.time())
+            id_token = tok['id_token']
+            token_type = tok['token_type']
+
+            # write new tokens
+
+            config.configuration.token.access_token = access_token
+            config.configuration.token.refresh_token = refresh_token
+            config.configuration.token.expires_in = expires_in
+            config.configuration.token.id_token = id_token
+            config.configuration.token.token_type = token_type
+            to_mqtt = json.dumps(config.__dict__, default=lambda o: o.__dict__)
+            socDebugLog("Config to MQTT:" + str(to_mqtt))
+            publish.single("openWB/set/vehicle/" + vehicle + "/soc_module/config",
+                           to_mqtt, retain=True, hostname="localhost")
+
+        else:
+            handleResponse("Refresh", ref.status_code, ref.text)
+
+    # call API for SoC
+    header = {'authorization': 'Bearer ' + access_token}
+    try:
+
+        req_soc = requests.get(soc_url, headers=header, verify=True)
+        # req_soc = requests.get(soc_url, headers=header, verify=True, timeout=req_timeout)
+
+    except Timeout:
+        socDebugLog("Soc Request Timed Out")
+
+    except RequestException:
+        socDebugLog("Soc Request Request Exception occured " + soc_url)
+
+    if Debug >= 1:
+        socDebugLog("SOC Request: " + str(req_soc.status_code))
+        socDebugLog("SOC Response: " + req_soc.text)
```

Comment:
```suggestion
    try:
        req_soc = req. req.get_http_session().get(soc_url, headers=header, verify=True)
    except Timeout:
        log.ecxeption("Soc Request Timed Out")
    except RequestException:
        log.exception("Soc Request Request Exception occured " + soc_url)
```
Entsprechend auch an den anderen Stellen.

Reply:
done